packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -917,4 +920,1044 @@ moduleFor('Components test: curly components', class extends RenderingTest {
+
+    this.runTask(() => this.rerender());
+
+    assert.equal(this.$('#inner-value').text(), '1', 'initial render of inner');
+    assert.equal(this.$('#middle-value').text(), '', 'initial render of middle (observers do not run during init)');
+
+    this.runTask(() => outer.set('value', 2));
+
+    assert.equal(this.$('#inner-value').text(), '2', 'second render of inner');
+    assert.equal(this.$('#middle-value').text(), '2', 'second render of middle');
+
+    this.runTask(() => outer.set('value', 3));
+
+    assert.equal(this.$('#inner-value').text(), '3', 'third render of inner');
+    assert.equal(this.$('#middle-value').text(), '3', 'third render of middle');
+
+    this.runTask(() => outer.set('value', 1));
+
+    assert.equal(this.$('#inner-value').text(), '1', 'reset render of inner');
+    assert.equal(this.$('#middle-value').text(), '1', 'reset render of middle');
+  }
+
+  ['@test non-block with each rendering child components']() {
+    this.registerComponent('non-block', {
+      template: strip`
+        In layout. {{#each items as |item|}}
+          [{{child-non-block item=item}}]
+        {{/each}}`
+    });
+
+    this.registerComponent('child-non-block', {
+      template: 'Child: {{item}}.'
+    });
+
+    let items = emberA(['Tom', 'Dick', 'Harry']);
+
+    this.render('{{non-block items=items}}', { items });
+
+    this.assertText('In layout. [Child: Tom.][Child: Dick.][Child: Harry.]');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('In layout. [Child: Tom.][Child: Dick.][Child: Harry.]');
+
+    this.runTask(() => this.context.get('items').pushObject('Sergio'));
+
+    this.assertText('In layout. [Child: Tom.][Child: Dick.][Child: Harry.][Child: Sergio.]');
+
+    this.runTask(() => this.context.get('items').shiftObject());
+
+    this.assertText('In layout. [Child: Dick.][Child: Harry.][Child: Sergio.]');
+
+    this.runTask(() => this.context.set('items', emberA(['Tom', 'Dick', 'Harry'])));
+
+    this.assertText('In layout. [Child: Tom.][Child: Dick.][Child: Harry.]');
+  }
+
+  ['@skip specifying classNames results in correct class'](assert) {
+    let clickyThing;
+
+    this.registerComponent('some-clicky-thing', {
+      ComponentClass: Component.extend({
+        tagName: 'button',
+        classNames: ['foo', 'bar'],
+        init() {
+          this._super(...arguments);
+          clickyThing = this;
+        }
+      })
+    });
+
+    this.render(strip`
+      {{#some-clicky-thing classNames="baz"}}
+        Click Me
+      {{/some-clicky-thing}}`
+    );
+
+    let expectedClassNames = ['ember-view', 'foo', 'bar', 'baz'];
+
+    assert.ok(this.$('button').is('.foo.bar.baz.ember-view'), `the element has the correct classes: ${this.$('button').attr('class')}`);
+    assert.deepEqual(clickyThing.get('classNames'), expectedClassNames, 'classNames are properly combined');
+    assert.deepEqual(this.$('button').attr('class').split(' '), expectedClassNames, 'all classes are set 1:1 in DOM');
```

Comment:
I think you can use `this.assertComponentElement` for this, e.g. https://github.com/emberjs/ember.js/blob/master/packages/ember-glimmer/tests/integration/components/curly-components-test.js#L157 (this might actually just be a duplicate of the new tests, if it's already covered there then we can consider :fire:-ing this)